examples/backend/src/ExampleAPI.hs

```diff
@@ -19,18 +19,19 @@ module ExampleAPI
 where
 
 import Data.Aeson (FromJSON, ToJSON)
+import Data.Maybe (fromMaybe)
 import Data.Text (Text)
 import Elm (ElmStreet (..))
 import Elm.Generic (Elm (..))
 import GHC.Generics (Generic)
 import Network.Wai (Application)
-import Servant (Get, JSON, Post, Proxy (..), ReqBody, Server, serve, type (:<|>) (..), type (:>))
+import Servant (Get, Handler, Header, JSON, Post, Proxy (..), ReqBody, Server, serve, type (:<|>) (..), type (:>))
 
 type UserAPI =
   "users" :> Get '[JSON] [User]
     :<|> "albert" :> Get '[JSON] User
     :<|> "signup" :> ReqBody '[JSON] User :> Post '[JSON] User
-    :<|> "sth" :> Post '[JSON] User
+    :<|> "sth" :> Header "someHeader" Text :> Post '[JSON] Text
```

Comment:
I think you should add another endpoint with a non-trivial to encode value (e.g. not `Int` or `Text`), so it shows how we decided deal with those for now.

I think we should also document things at this point, as this decision (and the reasoning behind it) isn't obvious.

Reply:
done